### DISCUSSION TITLE
List of cool projects by 42 students

### DISCUSSION THREAD

**Discussion by Fulguritude** [leeoocca/awesome-42#4] [Awesome projects]
I think it would be worthwhile to have a list of projects that were done by 42 students as inspiration material. I used to know a bunch of these, but I can't recall many of the students' names that would allow me to find them nowadays. So instead, I suggest that people interested for their own project (or someone else's) to appear on this page should nominate them here.

Info required:
  - project github address;
  - contributors' github login;
  - 42 student login;
  - a small description of why you think the project is interesting

Here is the list of those I remember (because I contributed to them)

 - https://github.com/LexouDuck/libccc/
   The current version of the `libft` that my brother (LexouDuck/aduquesn) and I (Fulguritude/tduquesn) have made combining the different additions we'd made to each of our own over our time at 42. It has since been largely, largely, largely extended. It is currently used in production for a cross-platform C communication client at our current company, and still evolving (recently, LexouDuck has been adding a cross-platform parser and generically-typed monads (in C, yes, you read that right)).
  
 - https://github.com/LexouDuck/RT/
   The RayTracer built by LexouDuck/aduquesn, Fulguritude/tduquesn, hbruvry/hbruvry, and Dr-Haas/ghaas. OpenCL support (with an option for choice of computational platform), all quadrics, MCPT, texturing, random scene generation, many debug features, and a full scene edition UI built by hand in a retro, pixel-art style over SDL.
   
 - https://github.com/Fulguritude/42_fractol
   A modest fract'ol over MinilibX. The thing that sets this fract'ol apart from the rest is that any polynomial formula can be chosen (not just `z^2 + c`), and events allow you to change the rendering formula at runtime. It also has a Mariani-Silver rendering mode, which is incredibly fast (`n*log(n)`).
   
 - https://github.com/Fulguritude/42_ft_printf
   A (comparatively) slow printf, but which implements full precision for printing floating-point values, using a homemade arbitrary precision arithmetic library. It also boasts a very large and useful test suite for printf, using "ft_asprintf" syntax to compare with the local implementation of asprintf.

**Comment by leeoocca** [comment]
I think it would be best to create a discussion topic for each project, that way people can upvote their favorite. What do you think?

**Reply by Fulguritude** [reply]
Hm. Maybe just take this discussion, and add a single comment thread per project ? If you think that's the right way to go about things, I'll edit the message above; remove each projects, and add each as a comment instead.

**Reply by leeoocca** [reply]
Sorry for the late reply. I think one per project or per project category (graphics, algorithms, etc) would be tidier, don't you think?

**Reply by Fulguritude** [reply]
I'm afraid having lots and lots of discussions might be hard to get around. But one discussion per project category seems like a good middle ground. The general README should probably add info explaining how to submit a project for the awesome list (with appropriate links to the relevant discussions).

As for categories: graphics, algorithmics, systems, networks, web, games, security, sound, robotics, haptic/VR/AR, general purpose libraries, devops, AI/ML, mathematics, pedagogy, language-specific ... anything else ? Should a project be present in multiple lists if it applies to multiple categories ?
